draft-ietf-wish-whep.md

```diff
@@ -633,6 +633,25 @@ data: /DA8AAAAAAAAAP///wb+06ACpQAmAiRDVUVJAACcHX//AACky4AMEERJU0NZTVdGMDQ1MjAwME
 ~~~~~
 {: title="scte35 example event"}
 
+### Events over WebRTC datachannel
+
+In some architectures sending events over a long polled HTTP request from a server that isn't the media server may be costly, such servers may support events over the webRTC datachannel.
+The protocol is identical to the above, with the following exceptions:
+
+- If both the SDP offer and the SDP answer contain an application section this indicates that Datachannel events are supported.
+~~~~~
+m=application 9 UDP/DTLS/SCTP webrtc-datachannel
+~~~~~
+- If the client then opens a reliable datachannel with a label that matches the URI of the sse Location: the server MUST send events as messages over this datachannel in the format described above.
+- The client may then listen for events on the datachannel instead of using the (similar) server-sent-events API
+
+This option can reduce resource consumption in 2 ways:
+- internal message passing within the server cluster since the events typically originate in the media server not the web server
```

Comment:
it could be argued that some events like viewercount, scte35 or reconnect are not generated by the media server but the business logic component.

Reply:
I think that depends on the selected architecture. If there are no efficiency benefits then the system shouldn't indicate support for datachannel based events.

Reply:
I hadn't considered mix/match event types over the 2 different transports. I thought of them as mutually exclusive, but if you think it useful please add some words...

Reply:
Honestly I'm not sure you'd ever do it... but its possible? Is the spec laid out currently so you can ask for multiple SSE streams of events for the one whep resource or is it a 1:1 relationship?

Reply:
Coming back to this.... to make an implementation easier for the WHEP server (but yes more complicated for the spec) you could specify that your SSE endpoint only allows events X Y and Z and the data channel implementation only allows events A B and C. Its entirely believable that someone may not want to allow for layer events from their SSE implementation because of all the reasons @steely-glint has outlined, but would accept viewercount events; as an example.